packages/state-transition/src/cache/epochCache.ts

```diff
@@ -403,8 +421,10 @@ export class EpochCache {
 
     return new EpochCache({
       config,
-      pubkey2index,
-      index2pubkey,
+      finalizedPubkey2index,
+      finalizedIndex2pubkey,
+      // `createFromState()` creates cache with empty unfinalizedPubkey2index. Be cautious to only pass in finalized state
```

Comment:
> Is it possible to programatically assert this condition? @tuyennhv does this break if we allow regen to read serialized states from disk?

Reply:
yes when we create an EpochCache from unfinalized state, this breaks the assumption. I don't know how to handle that case but think we should make it clear by changing method name to `createFromFinalizedState` and add this comment/assumption to the method header